query/src/main/scala/filodb/query/exec/ExecPlan.scala

```diff
@@ -502,19 +502,15 @@ abstract class NonLeafExecPlan extends ExecPlan {
       }
   }
 
-  /**
-   * Reduces the different ResultSchemas coming from each child to a single one.
-   * The default one here takes the first schema response, and checks that subsequent ones match the first one.
-   * Can be overridden if needed.
-   * @param rs the ResultSchema from previous calls to reduceSchemas / previous child nodes.  May be empty for first.
-   */
-  def reduceSchemas(rs: ResultSchema, resp: QueryResult): ResultSchema = {
-    resp match {
-      case QueryResult(_, schema, _, _, _, _) if rs == ResultSchema.empty =>
-        schema     /// First schema, take as is
-      case QueryResult(_, schema, _, _, _, _) =>
-        if (rs != schema) throw SchemaMismatch(rs.toString, schema.toString)
-        else rs
+  def reduceSchemas(r1: ResultSchema, r2: ResultSchema): ResultSchema = {
+    if (r1.isEmpty) r2
+    else if (r2.isEmpty) r1
+    else if (!r1.hasSameColumnsAs(r2))  {
+      throw SchemaMismatch(r1.toString, r2.toString)
+    } else {
+      val fixedVecLen = if (r1.fixedVectorLen.isEmpty && r2.fixedVectorLen.isEmpty) None
+      else Some(r1.fixedVectorLen.getOrElse(0) + r2.fixedVectorLen.getOrElse(0))
+      r1.copy(fixedVectorLen = fixedVecLen)
```

Comment:
(1) Ack'd this is pulled directly from existing code.
(2) I've never had a problem with it.

But since this is now the de-facto `reduceSchema`: are we guaranteed that all periodic plan `QueryResult` `ResultSchema`s will have their `fixedVectorLen`s defined if their results carry >0 rows? If not, this might accidentally truncate some rows.

Reply:
this code would not do any new harm if the `fixedVectorLen` is not defined for r2.
earlier code did not even account for `fixedVectorLen` of r2 as we were returning first not empty schema.